### PR TITLE
assignment list on dashboard

### PR DESCRIPTION
**Pull request by ndon01**

### CODE REVIEW

File: clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.html [hunk 1]
```diff
@@ -18,6 +18,6 @@ <h1 class="text-3xl font-bold text-on-surface">My Courses</h1>
   </div>
   <div class="w-full h-full bg-surface-c flex flex-col p-5 gap-10 overflow-y-scroll">
     <div *ngIf="myAssignments.length === 0"> No Assignments available </div>
-    <app-assignments-list [assignments]="myAssignments"></app-assignments-list>
+    <app-assignments-list *ngIf="myAssignments.length > 0" [assignments]="myAssignments"></app-assignments-list>
   </div>
```
Comment (ndon01) [inline clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.html:22]: *ngIf on this app-assignments-list lets us render this component only when we have assignments available from the network request

File: clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.ts [hunk 2]
```diff
@@ -32,10 +33,19 @@ export class DashboardPageComponent implements OnInit {
...
+        return assignments.map(assignment => {
+          if (typeof assignment.startDate === 'string') {
+            assignment.startDate = new Date(assignment.startDate);
+          }
+          if (typeof assignment.dueDate === 'string') {
+            assignment.dueDate = new Date(assignment.dueDate);
+          }
+          return assignment;
+        });
+      })).subscribe((data) =>
```
Comment (ndon01) [inline clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.ts:46]: String's aren't converted to Dates at runtime

Comment (SebastianAbarca) [reply clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.ts:46]: made a pipe checkout branch: SA-StringToDate

Comment (SebastianAbarca) [reply clms-angular/src/app/modules/dashboard/pages/dashboard-page/dashboard-page.component.ts:46]: ignore this. I didn't see you already did it

**Pull request merged**